dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/search/SavedSearchServiceTest.java

```diff
@@ -0,0 +1,58 @@
+package edu.harvard.iq.dataverse.search;
+
+import edu.harvard.iq.dataverse.DataverseRequestServiceBean;
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.EjbDataverseEngine;
+import edu.harvard.iq.dataverse.engine.command.impl.CreateSavedSearchCommand;
+import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
+import edu.harvard.iq.dataverse.persistence.dataverse.link.SavedSearch;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+
+@ExtendWith(MockitoExtension.class)
+class SavedSearchServiceTest {
+
+    @InjectMocks
+    private SavedSearchService savedSearchService;
+
+    @Mock
+    private DataverseSession dataverseSession;
+
+    @Mock
+    private DataverseRequestServiceBean dvRequestService;
+
+    @Mock
+    private EjbDataverseEngine commandEngine;
+
+    @Test
+    public void saveSavedDataverseSearch() {
```

Comment:
Do we have tests for publish and delete methods somewhere or are they missing?

Reply:
DataverseSaver didn't have unit tests so i just added Integration tests.